packages/replay/src/coreHandlers/handleGlobalEvent.ts

```diff
@@ -27,50 +37,20 @@ export function handleGlobalEventListener(replay: ReplayContainer): (event: Even
     }
 
     // Only tag transactions with replayId if not waiting for an error
-    // @ts-ignore private
-    if (!event.type || replay.recordingMode === 'session') {
+    if (isErrorEvent(event) || (isTransactionEvent(event) && replay.recordingMode === 'session')) {
       event.tags = { ...event.tags, replayId: replay.getSessionId() };
     }
 
-    // Collect traceIds in _context regardless of `recordingMode` - if it's true,
-    // _context gets cleared on every checkout
-    if (event.type === 'transaction' && event.contexts && event.contexts.trace && event.contexts.trace.trace_id) {
-      replay.getContext().traceIds.add(event.contexts.trace.trace_id as string);
-      return event;
-    }
-
-    // no event type means error
-    if (!event.type) {
-      replay.getContext().errorIds.add(event.event_id as string);
-    }
-
-    if (__DEBUG_BUILD__ && replay.getOptions()._experiments.traceInternals) {
+    if (__DEBUG_BUILD__ && replay.getOptions()._experiments.traceInternals && isErrorEvent(event)) {
       const exc = getEventExceptionValues(event);
       addInternalBreadcrumb({
         message: `Tagging event (${event.event_id}) - ${event.message} - ${exc.type}: ${exc.value}`,
       });
     }
 
-    // Need to be very careful that this does not cause an infinite loop
-    if (
-      replay.recordingMode === 'error' &&
-      event.exception &&
-      event.message !== UNABLE_TO_SEND_REPLAY // ignore this error because otherwise we could loop indefinitely with trying to capture replay and failing
-    ) {
-      setTimeout(async () => {
-        // Allow flush to complete before resuming as a session recording, otherwise
-        // the checkout from `startRecording` may be included in the payload.
-        // Prefer to keep the error replay as a separate (and smaller) segment
-        // than the session replay.
-        await replay.flushImmediate();
-
-        if (replay.stopRecording()) {
-          // Reset all "capture on error" configuration before
-          // starting a new recording
-          replay.recordingMode = 'session';
-          replay.startRecording();
-        }
-      });
+    if (afterSendHandler) {
+      // Pretend the error had a 200 response so we capture it
+      afterSendHandler(event, { statusCode: 200 });
```

Comment:
Can we add comments here about why we call `afterSendHandler` here?

Reply:
👍